orderbook/usecase/orderbook_usecase_test.go

```diff
@@ -0,0 +1,542 @@
+				TickId:         1,
+				Quantity:       "1000",
+				PlacedQuantity: "1500",
+				OrderDirection: "ask",
+				Etas:           "100",
+				PlacedAt:       "invalid", // Invalid timestamp
+			},
+			expectedError: &types.ParsingPlacedAtError{},
+			setupMocks: func(orderbookRepo *mocks.OrderbookRepositoryMock, tokensUsecase *mocks.TokensUsecaseMock) {
+				orderbookRepo.GetTickByIDFunc = getTickByIDFunc("100", 100, "ask")
+				tokensUsecase.GetSpotPriceScalingFactorByDenomFunc = func(baseDenom, quoteDenom string) (osmomath.Dec, error) {
+					return osmomath.NewDec(10), nil
+				}
+			},
+		},
+		{
+			name: "successful order processing",
+			order: orderbookdomain.Order{
+				TickId:         1,
+				OrderId:        1,
+				OrderDirection: "bid",
+				Owner:          "owner1",
+				Quantity:       "1000",
+				PlacedQuantity: "1500",
+				Etas:           "500",
+				ClaimBounty:    "10",
+				PlacedAt:       "1634764800000",
+			},
+			quoteAsset: orderbookdomain.Asset{Symbol: "ATOM", Decimals: 6},
+			baseAsset:  orderbookdomain.Asset{Symbol: "OSMO", Decimals: 6},
+			setupMocks: func(orderbookRepo *mocks.OrderbookRepositoryMock, tokensUsecase *mocks.TokensUsecaseMock) {
+				orderbookRepo.GetTickByIDFunc = getTickByIDFunc("500", 100, "bid")
+				tokensUsecase.GetSpotPriceScalingFactorByDenomFunc = func(baseDenom, quoteDenom string) (osmomath.Dec, error) {
+					return osmomath.NewDec(1), nil
+				}
+			},
+			expectedError: nil,
+			expectedOrder: orderbookdomain.LimitOrder{
+				TickId:           1,
+				OrderId:          1,
+				OrderDirection:   "bid",
+				Owner:            "owner1",
+				Quantity:         osmomath.NewDec(1000),
+				Etas:             "500",
+				ClaimBounty:      "10",
+				PlacedQuantity:   osmomath.NewDec(1500),
+				PlacedAt:         1634,
+				Price:            newDecFromStr("1.000001000000000000"),
+				PercentClaimed:   newDecFromStr("0.333333333333333333"),
+				TotalFilled:      newDecFromStr("600"),
+				PercentFilled:    newDecFromStr("0.400000000000000000"),
+				OrderbookAddress: "someOrderbookAddress",
+				Status:           "partiallyFilled",
+				Output:           newDecFromStr("1499.998500001499998500"),
+				QuoteAsset:       orderbookdomain.Asset{Symbol: "ATOM", Decimals: 6},
+				BaseAsset:        orderbookdomain.Asset{Symbol: "OSMO", Decimals: 6},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			// Create instances of the mocks
+			orderbookRepo := mocks.OrderbookRepositoryMock{}
+			tokensUsecase := mocks.TokensUsecaseMock{}
+
+			// Setup the mocks according to the test case
+			tc.setupMocks(&orderbookRepo, &tokensUsecase)
+
+			// Initialize the use case with the mocks
+			usecase := orderbookusecase.New(
+				&orderbookRepo,
+				nil,
+				nil,
+				&tokensUsecase,
+				nil,
+			)
+
+			// Call the method under test
+			result, err := usecase.CreateFormattedLimitOrder(tc.poolID, tc.order, tc.quoteAsset, tc.baseAsset, "someOrderbookAddress")
```

Comment:
nit: would be good to have a test case with the address varied for full coverage of the input set

Reply:
Addressed with [4e1e945](https://github.com/osmosis-labs/sqs/pull/492/commits/4e1e9457636004e1f5bf701246a10526abad37a4) :+1: